src/trio/testing/_raises_group.py

```diff
@@ -628,3 +906,64 @@ def expected_type(self) -> str:
                 raise AssertionError("unknown type")
         group_type = "Base" if self.is_baseexceptiongroup else ""
         return f"{group_type}ExceptionGroup({', '.join(subexcs)})"
+
+
+@t_final
+class NotChecked: ...
+
+
+class ResultHolder:
+    def __init__(
+        self,
+        expected_exceptions: tuple[ExpectedType[BaseExcT_co], ...],
+        actual_exceptions: Sequence[BaseException],
+    ) -> None:
+        self.results: list[list[str | type[NotChecked] | None]] = [
+            [NotChecked for _ in expected_exceptions] for _ in actual_exceptions
+        ]
+
+    def set_result(self, expected: int, actual: int, result: str | None) -> None:
+        self.results[actual][expected] = result
+
+    def get_result(self, expected: int, actual: int) -> str | None:
+        res = self.results[actual][expected]
+        assert res is not NotChecked
+        # why doesn't mypy pick up on the above assert?
+        return res  # type: ignore[return-value]
+
+    def get_all_results_for_actual(self, actual: int) -> Iterator[str | None]:
+        for res in self.results[actual]:
+            assert res is not NotChecked
+            yield res  # type: ignore[misc]
```

Comment:
[According to codecov](https://app.codecov.io/gh/python-trio/trio/pull/3145#359e2c748b0e9a4465cd5505f5b7214a-R934), this function is never executed.  Can we delete it?

Reply:
think so, will try to remember why I added it to make sure I didn't miss something in the transition